test/integration/zap-mint.ts

```diff
@@ -236,4 +236,108 @@ describe("Zap mint", () => {
     );
     await expectToFail(() => mintPromise, "ZapMint: Token not supported for zap minting");
   });
+
+  it("Should not mint supported token when collateral ratio < 100% and bdx was not provided", async () => {
+    const deployer = await getDeployer(hre);
+    const zapMintTestContract = (await hre.ethers.getContract("ZapMint", deployer)) as ZapMint;
+    const swapFrom = await getBdEu(hre);
+    const swapTo = await getWeth(hre);
+    const path = [swapFrom.address, swapTo.address];
+    const router = await getUniswapRouter(hre);
+    const decimals = await swapFrom.decimals();
+    const amountIn = numberToBigNumberFixed(1, decimals);
+    const currentBlock = await hre.ethers.provider.getBlock("latest");
+    const wethBdusPool = await getBDStableWethPool(hre, "BDUS");
+    await zapMintTestContract.addTokenSupportForMinting(swapFrom.address);
+
+    await swapFrom.approve(zapMintTestContract.address, amountIn);
+
+    const cr = 0.7;
+    await lockBdusCrAt(hre, cr);
+
+    const stableExpectedFromMint = await getBdUs(hre);
+    const mintPromise = zapMintTestContract.mintUniswapRouterOnly(
+      0,
+      0,
+      true,
+      stableExpectedFromMint.address,
+      wethBdusPool.address,
+      path,
+      amountIn,
+      router.address,
+      currentBlock.timestamp + 1e5
+    );
+
+    await expectToFail(() => mintPromise, "Not enough BDX inputted");
+  });
+
+  it("Should mint supported token when collateral ratio < 100% and bdx was provided", async () => {
+    const deployer = await getDeployer(hre);
+    const zapMintTestContract = (await hre.ethers.getContract("ZapMint", deployer)) as ZapMint;
+    const swapFrom = await getBdEu(hre);
+    const swapTo = await getWeth(hre);
+    const path = [swapFrom.address, swapTo.address];
+    const router = await getUniswapRouter(hre);
+    const decimals = await swapFrom.decimals();
+    const amountIn = numberToBigNumberFixed(0.01, decimals);
+    const currentBlock = await hre.ethers.provider.getBlock("latest");
+    const wethBdusPool = await getBDStableWethPool(hre, "BDUS");
+    await zapMintTestContract.addTokenSupportForMinting(swapFrom.address);
+
+    const bdx = await getBdx(hre);
+
+    const cr = 0.7;
+    await lockBdusCrAt(hre, cr);
+
+    const bdus = await getBdUs(hre);
+    const bdxInUsdPrice_d12 = await bdus.BDX_price_d12();
+    const wethInUsdPrice_d12 = await wethBdusPool.getCollateralPrice_d12();
+    const bdxPriceInWeth_d12 = BigNumber.from(1e12).mul(wethInUsdPrice_d12).div(bdxInUsdPrice_d12);
+
+    const bdxInMax = amountIn
+      .mul(to_d8(1 - cr))
+      .div(to_d8(cr))
+      .mul(bdxPriceInWeth_d12)
+      .div(1e12); // the remaining 30% of value
+
+    const stableExpectedFromMint = await getBdUs(hre);
+
+    await bdx.approve(zapMintTestContract.address, bdxInMax.mul(2));
+    await swapFrom.approve(zapMintTestContract.address, amountIn);
+
+    const bdeuBalanceBeforeMinting_d18 = await swapFrom.balanceOf(deployer.address);
```

Comment:
don't use deployer in tests when not necessary. use getUser1()

Reply:
done